ingest/indexer/service/client/pubsub_client.go

```diff
@@ -116,6 +119,21 @@ func (p *PubSubClient) PublishTokenSupplyOffset(ctx context.Context, tokenSupply
 	return p.publish(ctx, tokenSupplyOffset, p.tokenSupplyOffsetTopicId)
 }
 
+// PublishPair implements PubSubClient.PublishPair
+func (p *PubSubClient) PublishPair(ctx context.Context, pair indexerdomain.Pair) error {
+	// Check if project id and topic id are set
+	if p.projectId == "" || p.pairTopicId == "" {
+		return errors.New("project id and pool topic id must be set")
+	}
+
+	if pair.IdxDenom0 == pair.IdxDenom1 {
+		return fmt.Errorf("Publishing pool pairs with the same denom indexes %v", pair)
+	}
```

Comment:
Note: I've observed mysterious warning logs in data flow where these would be equal (invalid case)

However, breakpoints would never reach this branch